ffi/callbacks/callbacks_library/callbacks.c

```diff
@@ -0,0 +1,29 @@
+#include "callbacks.h"
+#include <unistd.h>
+#include <stdio.h>
+
+//
+// This is defined so the linker doesn't complain about missing main.
+// This is odd because a shared library (dylib/.so/etc.) shouldn't need a main.
+// The other examples define a main, too.
+//
+// That said, the CmakeLists.txt causes a callbacks_test executable to be
+// generated and runing that does print out "main called" message and exits.
+//
+int main() {
+  printf("main called\n");
+  return 1;
+}
+
+//
+// Our PerformCallback function.
+// It just loops forever and once per second calls the Dart callback function with
+// an ever increasing number.
+//
+void PerformCallback(Callback cb) {
+  int i = 0;
+  for (;;) {
+    cb(i++);
+    sleep(1);
```

Comment:
I completely agree about sleep() inside the C code.  My intent was to suggest that "some time later" the callback is being called.  The emphasis is on making the Dart code look async, which it does, IMO.

I can add a pthread and do the callback from within that, but again, I didn't want to add any code that distracts from doing the actual call passing callback from Dart to C and calling that callback from the C code.

I'd also have to look into what Windows provides as far as pthread goes.  It might just work, or I might have to add #if WINDOWS and #if LINUX kinds of blocks.

I appreciate that you approved the PR.  I'm willing to do whatever it takes to make this new example as ideal as possible.  Perhaps I can just add comments above the sleep() call with the explanation?

Reply:
>  The emphasis is on making the Dart code look async, which it does, IMO.

This is problematic. Because the Dart code is not asynchronous. It's synchronous but blocked.

> This is an example of passing a callback method from Dart to C and having that callback method called asynchronously.

This should be updated to synchronously.

I suggest to make this sample a sample of synchronous callbacks. WIth asynchronous callbacks you will have to use native ports instead of FFI callbacks. And if you want to keep this sample simple, asynchronous callbacks are not an option till we address https://github.com/dart-lang/sdk/issues/37022.